### ISSUE TITLE
Handling distribution-specific configuration

### ISSUE THREAD

**Issue by pellared** [signalfx/gdi-specification#198]
Currently the spec says:

https://github.com/signalfx/gdi-specification/blob/c9216b2d05faa82da4f5a62d801476994f56ff56/specification/configuration.md?plain=1#L15-L19

In practice, we see that this process it is not working. [Some of the configs in Java OTel distro](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/advanced-java-otel-configuration.html#metrics-collection-settings) are not defined in the GDI spec, not there are any GitHub issues to introduce them

What should we do?

1. Should we update the GDI configuration spec and introduce the env vars. Like e.g. here: https://github.com/signalfx/gdi-specification/pull/196
2. Should we change change the process to handle distribution-specific configuration?
3. (other proposals)

**Comment by pellared** [comment]
Personally, I propose to allow distributions to have their own config options in the form of `SPLUNK_{LANGUAGE}_{FEATURE}`.

This proposal is based on: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#language-specific-environment-variables

The current spec make sense for configuration that are not language specific. However, we see in practice that some of the configuration is tightly-coupled to some implementation-specific details.

**Comment by mateuszrzeszutek** [comment]
> In practice, we see that this process it is not working.

I think that the current process is unreasonably strict (and this is the reason why it's not being followed everywhere).

Take the profiler, for example: we first developed a POC implementation in Java, and only after we made sure it worked (and that other languages will need this feature too) we made the changes to the GDI spec. The profiler spec was changed several times when other language implementations (like .NET) were being developed.
The current process would require adding everything upfront, before introducing any change to any of the implementation libraries - this simply cannot work.

I think that the GDI spec and the process of adding new things to it should reflect the reality: any behavioral/configuration changes must be added to it IF we want to make them stable, or have more than one instrumentation library implement them.
To put it another way: if it's not in the GDI spec (or OTel spec), a feature cannot be considered stable and might change in backwards incompatible way at any point in time.

**Comment by pellared** [comment]
> The current process would require adding everything upfront, before introducing any change to any of the implementation libraries - this simply cannot work.

The spec say to have at least a GitHub issue upfront. I think it is a good as we may add the issue to a "milestone" and make sure it is addressed before doing a GDI release. We do not have to update the specification (design) upfront. I think it is just a matter of making sure that eventually things are landed in the GDI spec to make sure that other languages are adopting the same patterns.

Also take notice:

https://github.com/signalfx/gdi-specification/blob/c9216b2d05faa82da4f5a62d801476994f56ff56/specification/configuration.md?plain=1#L21-L30

WDYT?

**Comment by pjanotti** [comment]
The key is to have a discussion about why some env var is needed in some instrumentation: even if it becomes "stable" but makes only sense for one language/runtime I don't think we should bring it to the GDI spec.

**Comment by pellared** [comment]
Closing as stale.
